x-pack/legacy/plugins/monitoring/deprecations.js

```diff
@@ -26,6 +26,27 @@ export const deprecations = () => {
           `Config key "${CLUSTER_ALERTS_ADDRESS_CONFIG_KEY}" will be required for email notifications to work in 7.0."`
         );
       }
+
+      const fromPath = 'xpack.monitoring.elasticsearch';
+      const es = get(settings, 'elasticsearch');
+      if (es) {
+        if (es.username === 'elastic') {
+          log(
+            `Setting [${fromPath}.username] to "elastic" is deprecated. You should use the "kibana" user instead.`
+          );
+        }
+        if (es.ssl) {
+          if (es.ssl.key !== undefined && es.ssl.certificate === undefined) {
+            log(
+              `Setting [${fromPath}.ssl.key] without [${fromPath}.ssl.certificate] is deprecated. This has no effect, you should use both settings to enable TLS client authentication to Elasticsearch.`
+            );
+          } else if (es.ssl.certificate !== undefined && es.ssl.key === undefined) {
+            log(
+              `Setting [${fromPath}.ssl.certificate] without [${fromPath}.ssl.key] is deprecated. This has no effect, you should use both settings to enable TLS client authentication to Elasticsearch.`
+            );
+          }
+        }
+      }
```

Comment:
I think this code should be its own array item, since it's not related to the adjacent item above it

Reply:
Good idea, I'll split it out.

Reply:
Good idea, I'll split it out.